Libraries/oneMKL/matrix_mul_mkl/matrix_mul_mkl_bf16.cpp

```diff
@@ -0,0 +1,83 @@
+// Standard SYCL header
+#include <CL/sycl.hpp>
+// STL classes
+#include <exception>
+#include <iostream>
+// Declarations for Intel oneAPI Math Kernel Library SYCL/DPC++ APIs
+#include "oneapi/mkl.hpp"
+int main(int argc, char *argv[]) {
+
+	// User obtains data here for A, B, C matrices, along with setting m, n, k, ldA, ldB, ldC.
+    // For this example, A, B and C should be initially stored in a std::vector,
+    //   or a similar container having data() and size() member functions.
+
+using namespace oneapi::mkl;
+int64_t m = 10, n = 6, k = 8, ldA = 12, ldB = 8, ldC = 10;
+int64_t sizea = ldA * k, sizeb = ldB * n, sizec = ldC * n;
+double alpha = 1.0, beta = 0.0;
+// Allocate matrices
+std::vector<bfloat16> A(sizea);
+std::vector<bfloat16> B(sizeb);
+std::vector<bfloat16> C(sizec);
+// Initialize matrices […]
+std::fill(A.begin(), A.end(), 0);
+std::fill(B.begin(), B.end(), 0);
+std::fill(C.begin(), C.end(), 0);
+
+
+    sycl::device my_device = sycl::device(sycl::default_selector());
+    // Create asynchronous exceptions handler to be attached to queue.
+    // Not required; can provide helpful information in case the system isn’t correctly configured.
+    auto my_exception_handler = [](sycl::exception_list exceptions) {
+        for (std::exception_ptr const& e : exceptions) {
+            try {
+                std::rethrow_exception(e);
+            }
+            catch (sycl::exception const& e) {
+                std::cout << "Caught asynchronous SYCL exception:\n"
+                    << e.what() << std::endl;
+            }
+            catch (std::exception const& e) {
+                std::cout << "Caught asynchronous STL exception:\n"
+                    << e.what() << std::endl;
+            }
+        }
+    };
+    // create execution queue with exception handler attached
+    sycl::queue my_queue(my_device, my_exception_handler);
+    // create sycl buffers of matrix data for offloading between device and host
+    sycl::buffer<bfloat16, 1> A_buffer(A.data(), A.size());
+    sycl::buffer<bfloat16, 1> B_buffer(B.data(), B.size());
+    sycl::buffer<bfloat16, 1> C_buffer(C.data(), C.size());
+    // add oneapi::mkl::blas::gemm to execution queue and catch any synchronous exceptions
+    try {
+        using oneapi::mkl::blas::gemm;
+        using oneapi::mkl::transpose;
+        gemm(my_queue, transpose::nontrans, transpose::nontrans, m, n, k, alpha, A_buffer, ldA, B_buffer,
+           ldB, beta, C_buffer, ldC);

```

Comment:
`my_queue.wait_and_throw()` should be placed here inside the `try` so any execution-related exceptions can be handled appropriately.